src/expr/src/expr/expr_array_to_string.rs

```diff
@@ -100,7 +111,8 @@ fn array_to_string(
     mut writer: &mut dyn Write,
 ) {
     let mut first = true;
-    for element in array.iter().flatten() {
+    for element in array.flatten() {
+        let Some(element) = element else { continue };
```

Comment:
The old `.iter().flatten()` means to iter 1 layer, then remove nulls.

The new `.flatten()` means to flatten (unnest) recursively. Nulls are removed via let-else.

The 2 `flatten` are completely different, one on `Iterator<Item=Option<T>>` while the other on `ListRef`